libs/barista-components/sunburst/src/sunburst.ts

```diff
@@ -0,0 +1,261 @@
+  selectedRelativeValue: number;
+  labelAsAbsolute: boolean = true;
+
+  @HostListener('click', ['$event']) onClick(ev?: MouseEvent): void {
+    this.select(ev);
+  }
+
+  @Input() series: DtSunburstNode[];
+  @Input() selected: DtSunburstNode[];
+  @Input() noSelectionLabel: string = 'All';
+  @Input() valueDisplayMode: DtSunburstValueMode = DtSunburstValueMode.ABSOLUTE;
+
+  @Output() selectedChange: EventEmitter<DtSunburstNode[]> = new EventEmitter();
+
+  /** Template portal for the default overlay */
+  // tslint:disable-next-line: use-default-type-parameter no-any
+  private _portal: TemplatePortal<any> | null;
+
+  /** Reference to the open overlay. */
+  private _overlayRef: OverlayRef;
+
+  constructor(
+    private _overlayService: Overlay,
+    private _viewContainerRef: ViewContainerRef,
+    private _elementRef?: ElementRef<HTMLElement>,
+    @Optional()
+    @Inject(DT_UI_TEST_CONFIG)
+    private _config?: DtUiTestConfiguration,
+  ) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.series?.currentValue !== changes.series?.previousValue) {
+      this.filledSeries = fillNodes(this.series);
+      this.render();
+    }
+
+    if (changes.selected?.currentValue !== changes.selected?.previousValue) {
+      this._selected = getSelectedNodesFromOutside(
+        this.filledSeries,
+        this.selected,
+      );
+      this.render();
+    }
+
+    if (
+      changes.valueDisplayMode?.currentValue !==
+      changes.valueDisplayMode?.previousValue
+    ) {
+      this.labelAsAbsolute =
+        this.valueDisplayMode === DtSunburstValueMode.ABSOLUTE;
+    }
+  }
+
+  select(event?: MouseEvent, slice?: DtSunburstSlice): void {
+    event?.stopPropagation();
+
+    if (slice) {
+      this._selected = getSelectedNodes(this.filledSeries, slice.data);
+
+      this.selectedChange.emit(this._selected.map(node => node.origin));
+    } else {
+      this._selected = [];
+
+      this.selectedChange.emit([]);
+    }
+
+    this.render();
+  }
+
+  render(): void {
+    const nodesWithState = getNodesWithState(
+      this.filledSeries,
+      getSelectedId(this.filledSeries, this._selected),
+    );
+    this.slices = getSlices(nodesWithState);
+
+    if (this._selected && this._selected.length) {
+      this.selectedLabel = this._selected.slice(-1)[0].label ?? '';
+      this.selectedValue = this._selected.slice(-1)[0].value ?? 0;
+      this.selectedRelativeValue =
+        this._selected.slice(-1)[0].valueRelative ?? 0;
+    } else {
+      this.selectedLabel = this.noSelectionLabel;
+      this.selectedValue = getValue(this.filledSeries);
+      this.selectedRelativeValue = 1;
+    }
+  }
+  /** Programmatically close the overlay. */
+  closeOverlay(): void {
+    this._dismissOverlay();
+  }
+
+  /** Puts the overlay in place. */
+  openOverlay(node: DtSunburstSlice): void {
+    const origin = this.getOriginFromSlice(node);
+
+    this._dismissOverlay();
+    this._createOverlay(origin, node.data);
+  }
```

Comment:
If this should be exposed to the consumer, please add it to the methods available on the sunburst instance in the readme.